google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -143,118 +180,91 @@ def create(
 
-        if bool(bq_source) and not is_tabular_dataset_metadata:
-            raise ValueError(
-                "A tabular metadata_schema uri must be used when using a BigQuery source"
-            )
+        # Validate metadata_schema_uri and import_schema_uri for Dataset class
+        cls._validate_import_schema_uri(import_schema_uri)
 
-        api_client = cls._instantiate_client(location=location, credentials=credentials)
+        if metadata_schema_uri == schema.dataset.metadata.tabular:
+            datasource = datasources.TabularDatasource(gcs_source, bq_source)
+        else:
+            datasource = datasources.NonTabularDatasource()
+            if import_schema_uri:
+                datasource = datasources.NonTabularDatasourceImportable(
+                    gcs_source, import_schema_uri, data_item_labels
+                )
 
-        # If this is tabular enrich the dataset metadata with source
-        dataset_metadata = {}
-        if is_tabular_dataset_metadata:
-            if gcs_source:
-                dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
-            elif bq_source:
-                dataset_metadata = {
-                    "input_config": {"bigquery_source": {"uri": bq_source}}
-                }
-
-        return cls._create_and_import(
-            api_client=api_client,
+        return cls._create_encapsulated(
             display_name=display_name,
-            parent=initializer.global_config.common_location_path(
-                project=project, location=location
-            ),
             metadata_schema_uri=metadata_schema_uri,
-            dataset_metadata=dataset_metadata,
-            request_metadata=metadata,
+            datasource=datasource,
             labels=labels,
-            project=project or initializer.global_config.project,
-            location=location or initializer.global_config.location,
-            credentials=credentials or initializer.global_config.credentials,
-            gcs_source=gcs_source,
-            import_schema_uri=import_schema_uri,
-            data_items_labels=data_items_labels,
+            project=project,
+            location=location,
+            credentials=credentials,
+            request_metadata=request_metadata,
             sync=sync,
         )
 
     @classmethod
     @base.optional_sync()
-    def _create_and_import(
+    def _create_encapsulated(
         cls,
-        api_client: DatasetServiceClient,
         display_name: str,
-        parent: str,
         metadata_schema_uri: str,
-        dataset_metadata: str,
-        request_metadata: str,
-        project: str,
-        location: str,
-        credentials: Optional[auth_credentials.Credentials],
-        labels: Optional[Dict] = None,
-        gcs_source: Optional[Sequence[str]] = None,
-        import_schema_uri: Optional[str] = None,
-        data_items_labels: Optional[Dict] = None,
+        datasource: Union[
```

Comment:
Just have them pass the `Datasource` abstract class.

Reply:
Good point. Thanks for the feedback @ivanmkc